tests/queries/0_stateless/02960_alter_table_part_query_parameter.sql

```diff
@@ -0,0 +1,15 @@
+drop table if exists data;
+create table data (key Int) engine=MergeTree() order by key;
+
+insert into data values (1);
+
+set param_part='all_1_1_0';
+alter table data detach part {part:String};
+alter table data attach part {part:String};
+set param_part='all_2_2_0';
+alter table data detach part {part:String};
+alter table data drop detached part {part:String} settings allow_drop_detached=1;
```

Comment:
BTW, there is `Identifier` parameter type, may be it is a better fit here.

Reply:
DROP PART accept only string literal